### PR TITLE
Add official support for the ST M24 I2C EEPROM series

### PR DESCRIPTION
**Pull request by benediktibk**
It is already with the existing implementation of the AT24 possible to use an EEPROM from the ST M24 series, although it is not very well documented (see #73064 and #69021). To avoid this confusion I have generalized the driver for the AT24 and AT25 so that it can be reused for other compatibles and added specific bindings for the ST M24 I2C automotive series.
With this approach it is rather clear which ICs are supported by which driver, and it can be easily extended for other compatible ICs.

### CODE REVIEW

File: dts/bindings/mtd/st,m24c04-a125.yaml
Comment (erwango) [inline dts/bindings/mtd/st,m24c04-a125.yaml:16]: Do we actually need to have all these "non-compatible" compatibles using fixed properties?

What about a more generic `st,m24xxx` compatible ?
Aim it to document that a class of devices is supported and show how to configure it.

File: drivers/eeprom/eeprom_at2x.c
Comment (erwango) [inline drivers/eeprom/eeprom_at2x.c]: This is a lot of effort to duplicate common code.  There should be a better option.

If the problem is to use the right `DT_DRV_COMPAT`, note that it's possible to define multiple compatibles on a node.
For instance:
```
	compatible = "st,m24c64-a125", "atmel,at2x";
```
(dt spec mentions this should go from the most specific to the most generic).
Then , you can use `#define DT_DRV_COMPAT atmel_at2x`

**Review by erwango — COMMENTED** [review]
Thanks for digging into this direction, but adequate use of dts should help getting things more simple.

File: drivers/eeprom/eeprom_at2x.c
Comment (pdgendt) [inline drivers/eeprom/eeprom_at2x.c:685]: No need to create `api` instances for each compatible, just re-use a single `spi` or `i2c` instance.

Comment (benediktibk) [reply drivers/eeprom/eeprom_at2x.c:685]: good point

File: dts/bindings/mtd/st,m24c04-a125.yaml
Comment (benediktibk) [reply dts/bindings/mtd/st,m24c04-a125.yaml:16]: I would rather prefer to be specific, as this has multiple advantages:

1. If I only document the proper usage, I still have to add some text which has to be maintained. And usually documentation gets forgotten during updates and changes and it is therefore more likely be incorrect in the future.
2. This adds a build time check, which avoids improper use and therefore frustration and bugs in real applications. And as the hardware won't change anymore I think this is worth the effort.
3. Having a very generic compatible brought us to the current situation. From my experience it is rather handy on the first sight to add a more generic compatible, but in the end hardware vendors tend to do all kind of weird stuff with their hardware identifiers. There might not be something currently in the range of st,m24xxx which is incompatible with this driver, but I wouldn't bet on this to be true in the future as well.
4. It is more clear to a user which compatible should be used for which IC in the hardware design.

File: drivers/eeprom/eeprom_at2x.c
Comment (benediktibk) [reply drivers/eeprom/eeprom_at2x.c]: Thanks to the input of @pdgendt the code duplication is now down to almost none:

    #define DT_DRV_COMPAT atmel_at25
    DT_INST_FOREACH_STATUS_OKAY_VARGS(EEPROM_AT2X_INST_DEFINE, DT_DRV_COMPAT, EEPROM_AT2X_SPI_BUS,
                      EEPROM_AT2X_BUS_API_SPI, ASSERT_AT25_ADDR_W_VALID)
    #undef DT_DRV_COMPAT

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]
NAK. This will become a maintenance nightmare given all the different, compatible EEPROMs out there. I wish you had approached me before embarking on this.

**Comment by benediktibk** [comment]
> NAK. This will become a maintenance nightmare given all the different, compatible EEPROMs out there. I wish you had approached me before embarking on this.

Is the NACK for only the IC specific compatibles, or the complete approach? Because I wouldn't be a big fan of having a more generic st,m24xxx (as explained here https://github.com/zephyrproject-rtos/zephyr/pull/73136#discussion_r1609725955), but I would consider this still be better than the current status.

**Comment by henrikbrixandersen** [comment]
> Is the NACK for only the IC specific compatibles, or the complete approach? Because I wouldn't be a big fan of having a more generic st,m24xxx (as explained here [#73136 (comment)](https://github.com/zephyrproject-rtos/zephyr/pull/73136#discussion_r1609725955)), but I would consider this still be better than the current status.

I really do not see an issue with the current compatible and I would not want to add a bunch of vendor-specific aliases for this.

As mentioned earlier, I believe this can be solved by improving the documentation on how to configure serial EEPROMs in Zephyr. I plan on starting this work prior to v3.7.0.

**Comment by benediktibk** [comment]
> > Is the NACK for only the IC specific compatibles, or the complete approach? Because I wouldn't be a big fan of having a more generic st,m24xxx (as explained here [#73136 (comment)](https://github.com/zephyrproject-rtos/zephyr/pull/73136#discussion_r1609725955)), but I would consider this still be better than the current status.
> 
> I really do not see an issue with the current compatible and I would not want to add a bunch of vendor-specific aliases for this.
> 
> As mentioned earlier, I believe this can be solved by improving the documentation on how to configure serial EEPROMs in Zephyr. I plan on starting this work prior to v3.7.0.

Unfortunately I will have to disagree, because as a user it is rather confusing to use the compatible of A for B. I'm not convinced that documentation will solve this issue. On top of that, having build time checks and therefore a mechanism to fail early avoids a lot of issues in the long run.

**Comment by pdgendt** [comment]
I like [the comment](https://github.com/zephyrproject-rtos/zephyr/pull/73136#discussion_r1609637588) of @erwango, put a generic compatible at the end if the list?

File: tests/drivers/build_all/eeprom/app.overlay
Comment (benediktibk) [inline tests/drivers/build_all/eeprom/app.overlay:55]: @pdgendt @erwango Did you mean like this, and leaving the additions to eeprom_at2x.c out?

**Comment by benediktibk** [comment]
But anyway, @henrikbrixandersen is strongly against this, he prefers to stick to only the historical binding of atmel,at24 as it preceeds the reimplementation by other vendors and wants to avoid the confusion with additional documentation. Please correct me, if I am (unintentionally) citing you incorrectly.

**Comment by henrikbrixandersen** [comment]
> But anyway, @henrikbrixandersen is strongly against this, he prefers to stick to only the historical binding of atmel,at24 as it preceeds the reimplementation by other vendors and wants to avoid the confusion with additional documentation. Please correct me, if I am (unintentionally) citing you incorrectly.

No, that is correct.

**Comment by erwango** [comment]
> > But anyway, @henrikbrixandersen is strongly against this, he prefers to stick to only the historical binding of atmel,at24 as it preceeds the reimplementation by other vendors and wants to avoid the confusion with additional documentation. Please correct me, if I am (unintentionally) citing you incorrectly.
> 
> No, that is correct.

@henrikbrixandersen I understand your position, but what would you propose to explicit the fact that `atmel,at24` compatible devices are supported? 
This is already the second PR we have on the same topic and I guess that the ones pushing PRs are much less than the one that needed it and gave up. Obviously we miss some way to document this clearly.

**Comment by pdgendt** [comment]
I tend to agree, there are ongoing discussions whether to rename `struct net_buf` to something generic, which has a much bigger impact than to generalize a common driver for different peripherals.

**Comment by henrikbrixandersen** [comment]
> I tend to agree, there are ongoing discussions whether to rename `struct net_buf` to something generic, which has a much bigger impact than to generalize a common driver for different peripherals.

I honestly don't believe the two are comparable. Also, as you may recall, it was decided not to rename net_buf for now, but merely move it up to increase visibility.

**Comment by erwango** [comment]
> > @henrikbrixandersen I understand your position, but what would you propose to explicit the fact that `atmel,at24` compatible devices are supported?
> > This is already the second PR we have on the same topic and I guess that the ones pushing PRs are much less than the one that needed it and gave up. Obviously we miss some way to document this clearly.
> 
> My plan is to add a documentation page under https://docs.zephyrproject.org/latest/hardware/peripherals/eeprom.html detailing how to look up the various properties of a serial EEPROM in the data sheet with examples of how to represent these in a devicetree node.
> 
> Furthermore, I plan revisiting all occurrences of `compatible = "atmel,at2*"` within the tree and add a more specific compatible to the left of it (sort of like what is done here, but with the specific part number instead of `xxx`: https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/shields/x_nucleo_eeprma2/x_nucleo_eeprma2.overlay). This should help people grepping through the tree to realise that these devices are already supported.
> 
> We can also extend the help text of the bindings and Kconfig options to list "st,m24xxx" and other families to help people find them.

Ok, then, that's basically quite aligned with the content of my review ([here](https://github.com/zephyrproject-rtos/zephyr/pull/73136#discussion_r1609624013) and [here](https://github.com/zephyrproject-rtos/zephyr/pull/73136#discussion_r1609637588)) so I can't agree more.

**Comment by benediktibk** [comment]
Lets see if I have understood it correctly. This PR now contains only additional bindings and build_all tests which show how they can be used. This doesn't impact the implementation of the AT24 at all, but states explicitly which devices are supported, with the added benefit of build time failures for improper usage.

**Review by decsny — DISMISSED** [review]
I don't want to start the practice of replacing DTS property entries with dozens of const specifying compatibles

According to DT spec, the compatible specifies the programming model of the device. I.e., the binding is not the place for entering property values of DT, that is why we have devicetree source files with syntax to do that. All of these devices, if they have the same programming model, do not need new compatibles just to avoid actually entering the property values in the DTS.

With that being said, we *can* match multiple compatibles against the same device driver and I am in favor to make maybe just a couple, one for each vendor's eeprom family, that can match against this driver, for the clarification of support. But if we are adding dozens to just put property values it's too much.

**Comment by decsny** [comment]
@henrikbrixandersen if you are saying we shouldn't make multiple compatibles for the same driver that are just "aliases", I think that is wrong, it is clearly supported and I would even say encouraged by DT spec. Calling them aliases is very misleading, they ARE different devices, they just can use the same driver.

It might be nice to add in the future though some macros for a device driver to iterate a list of compatibles rather than just one instead of having multiple DT_FOREACH for each compatible

**Comment by henrikbrixandersen** [comment]
> @henrikbrixandersen if you are saying we shouldn't make multiple compatibles for the same driver that are just "aliases", I think that is wrong, it is clearly supported and I would even say encouraged by DT spec. Calling them aliases is very misleading, they ARE different devices, they just can use the same driver.

No, I am saying we shouldn't add multiple _bindings_ which are just aliases of each other with const properties (as is done in this PR).

**Comment by decsny** [comment]
> > @henrikbrixandersen if you are saying we shouldn't make multiple compatibles for the same driver that are just "aliases", I think that is wrong, it is clearly supported and I would even say encouraged by DT spec. Calling them aliases is very misleading, they ARE different devices, they just can use the same driver.
> 
> No, I am saying we shouldn't add multiple _bindings_ which are just aliases of each other with const properties (as is done in this PR).

Well, to clarify, I am saying we should add a few bindings which will be identical except for the compatible string, which is similar to what this PR is doing, except just no const properties and the compatibles should I think be just one per vendor's eeprom family that uses the driver.

**Comment by henrikbrixandersen** [comment]
> Well, to clarify, I am saying we should add a few bindings which will be identical except for the compatible string, which is similar to what this PR is doing, except just no const properties and the compatibles should I think be just one per vendor's eeprom family that uses the driver.

I don't see how that aligns with your previous comment:

> According to DT spec, the compatible specifies the programming model of the device.

The programming model for these devices, vendor and model aside, is the AT24/AT25 model initially developed by Atmel.

**Comment by decsny** [comment]
> > Well, to clarify, I am saying we should add a few bindings which will be identical except for the compatible string, which is similar to what this PR is doing, except just no const properties and the compatibles should I think be just one per vendor's eeprom family that uses the driver.
> 
> I don't see how that aligns with your previous comment:
> 
> > According to DT spec, the compatible specifies the programming model of the device.
> 
> The programming model for these devices, vendor and model aside, is the AT24/AT25 model initially developed by Atmel. 
> 

Okay, I'll admit the word "specifies" was a bit ambiguous a best  and inaccurate at worst. Specifically what I meant is that the *purpose* of a compatible is to be a way to *determine* a programming model for a device.

The format of a compatible is recommended to be vendor,model-number according to DT spec. As far as I know, the atmel AT2x are model numbers of parts vended by atmel. And therefore the current binding is valid.

And as far as ai understand, the situation here is that the other eeproms mentioned by this PR are another vendor's family of eeproms with a different model number that fits a similar programming model as the atmel AT2x. Therefore, these devices can use the same driver.

If my understanding is correct and that is the case, I think we should make a compatible for each of these other vendors eeprom families. We don't need one for each and every part number like this PR currently has, but it's not accurate to call an ST m2 whatever eeprom an atmel AT24. Just because they have a similar programming model doesn't mean they are the same device. And the author already listed some of the practical benefits of the compatible being accurately descriptive so I won't repeat those.

**Comment by benediktibk** [comment]
@decsny Would you prefer having no real driver behind for instance an st,m24xxx and just adding the at,at24 additionally in the devicetree, or rather have an actual driver and specify only one compatible st,m24xxx? The driver in this case of course would be the same one, just with another DT_FOREACH call.

**Comment by de-nordic** [comment]
This is not a good idea.
First of all the automotive specification does not touch the software properties of the device nor the protocol, these are variants of devices tested and approved for specific use, but these are physical properties being tested.
All the properties that are important for a driver are already covered by the generic definition; since there are no driver changes here, which enable specific feature for some specific device model, there is no point to have compatible for it - generic code covers it.
Note also that that there area already 8 bindings for automotive, but there is none for base devices (non-automotive), which would add next 8, now you have 16. If there are military, you have up to another 8, if there are radiation hardened - up to next 8... extended temperature range - next 8. Now you have 40?

It often happens that you work in prototyping, before env tests, with cheaper variant of device (the ordinary 85C range), and then can switch to the final device and complete tests. Then you have to change DTS, but nothing really changes in code.

What if your supplier offers you different vendor, same device, same characteristics. Do you change DTS? Wouldn't it be confusing to have different device in DTS and on board?

I think the confusion is too often used as an argument. We just lack time to pay attention.

**Comment by decsny** [comment]
> @decsny Would you prefer having no real driver behind for instance an st,m24xxx and just adding the at,at24 additionally in the devicetree, or rather have an actual driver and specify only one compatible st,m24xxx? The driver in this case of course would be the same one, just with another DT_FOREACH call.

I meant to just put one string in DT but otherwise yes this is what I meant

**Comment by decsny** [comment]
> This is not a good idea.
> First of all the automotive specification does not touch the software properties of the device nor the protocol, these are variants of devices tested and approved for specific use, but these are physical properties being tested.
> All the properties that are important for a driver are already covered by the generic definition; since there are no driver changes here, which enable specific feature for some specific device model, there is no point to have compatible for it - generic code covers it.
> Note also that that there area already 8 bindings for automotive, but there is none for base devices (non-automotive), which would add next 8, now you have 16. If there are military, you have up to another 8, if there are radiation hardened - up to next 8... extended temperature range - next 8. Now you have 40?
> 
> It often happens that you work in prototyping, before env tests, with cheaper variant of device (the ordinary 85C range), and then can switch to the final device and complete tests. Then you have to change DTS, but nothing really changes in code.
> 
> What if your supplier offers you different vendor, same device, same characteristics. Do you change DTS? Wouldn't it be confusing to have different device in DTS and on board?
> 
> I think the confusion is too often used as an argument. We just lack time to pay attention.

The thing is, you can use any of the strings. You don't have to change it, it's not like a user has to publish their DT to be graded for accuracy of all the strings by the zephyr community. So I don't see a problem with having some more binding for clarity.

It would be interesting if we can upgrade our DT tooling a bit to be able to specify multiple compable strings in one binding, instead of making a lot of different files for one string each, this seems to be how Linux handles situations like this: https://www.kernel.org/doc/Documentation/devicetree/bindings/eeprom/at24.yaml

**Comment by decsny** [comment]
> > This is not a good idea.
> > First of all the automotive specification does not touch the software properties of the device nor the protocol, these are variants of devices tested and approved for specific use, but these are physical properties being tested.
> > All the properties that are important for a driver are already covered by the generic definition; since there are no driver changes here, which enable specific feature for some specific device model, there is no point to have compatible for it - generic code covers it.
> > Note also that that there area already 8 bindings for automotive, but there is none for base devices (non-automotive), which would add next 8, now you have 16. If there are military, you have up to another 8, if there are radiation hardened - up to next 8... extended temperature range - next 8. Now you have 40?
> > 
> > It often happens that you work in prototyping, before env tests, with cheaper variant of device (the ordinary 85C range), and then can switch to the final device and complete tests. Then you have to change DTS, but nothing really changes in code.
> > 
> > What if your supplier offers you different vendor, same device, same characteristics. Do you change DTS? Wouldn't it be confusing to have different device in DTS and on board?
> > 
> > I think the confusion is too often used as an argument. We just lack time to pay attention.
> 
> The thing is, you can use any of the strings. You don't have to change it, it's not like a user has to publish their DT to be graded for accuracy of all the strings by the zephyr community. So I don't see a problem with having some more compatibles for clarity and accuracy.

I was thinking also it would be interesting if we can upgrade our DT tooling a bit to be able to specify multiple compable strings in one binding, instead of making a lot of different files for one string each, this seems to be how Linux handles situations like this: https://www.kernel.org/doc/Documentation/devicetree/bindings/eeprom/at24.yaml

**Comment by erwango** [comment]
> Note also that that there area already 8 bindings for automotive, but there is none for base devices (non-automotive), which would add next 8, now you have 16. If there are military, you have up to another 8, if there are radiation hardened - up to next 8... extended temperature range - next 8. Now you have 40?

There is no need to provide a binding for each reference. Having only one vendor specific, part generic, `st,m24xxx` compatible would be sufficient to establish that the part is supported.

**Comment by decsny** [comment]
> > Note also that that there area already 8 bindings for automotive, but there is none for base devices (non-automotive), which would add next 8, now you have 16. If there are military, you have up to another 8, if there are radiation hardened - up to next 8... extended temperature range - next 8. Now you have 40?
> 
> There is no need to provide a binding for each reference. Having only one vendor specific, part generic, `st,m24xxx` compatible would be sufficient to establish that the part is supported.

This is what I'm suggesting although technically I don't think it aligns with the DT spec recommendations perfectly, we don't have scalable tooling with regex pattern matching for compatibles like Linux right now, so this is probably the most reasonable option for us in zephyr at the moment

**Comment by de-nordic** [comment]
> The thing is, you can use any of the strings. You don't have to change it, it's not like a user has to publish their DT to be graded for accuracy of all the strings by the zephyr community. 

Yeah, that is true.

But note also how using the generic description, instead of read-only binding, you can substitute bigger parts for lower.
For example you can have
```
  compatible =  `atmel,at24c04`, `atmel,at24c08`;
```
with size set for the smaller one, if you can not get the smaller one but you can swallow the cost and the fact that you will only use half of the bigger one, till you restock, number of available devices "in case" increases.

**Comment by de-nordic** [comment]
> 
> There is no need to provide a binding for each reference. Having only one vendor specific, part generic, `st,m24xxx` compatible would be sufficient to establish that the part is supported.

OK, maybe. I do not want to state that it is unimportant for vendors to use their naming. `st,m24xxx` thing may be ok, but that is not what the PR is doing at this point.
The PR is very specific even to the point where it defines bindings for parts that are "(...) compliant with the very high level of reliability defined by the Automotive standard AEC-Q100 grade 1.", that is the -`a125` suffix, which does not change that this is sitll `st,m24xxx`. I do not think that even the DTS from kernel with regexes is handling that.`

**Comment by decsny** [comment]
> > There is no need to provide a binding for each reference. Having only one vendor specific, part generic, `st,m24xxx` compatible would be sufficient to establish that the part is supported.
> 
> OK, maybe. I do not want to state that it is unimportant for vendors to use their naming. `st,m24xxx` thing may be ok, but that is not what the PR is doing at this point. The PR is very specific even to the point where it defines bindings for parts that are "(...) compliant with the very high level of reliability defined by the Automotive standard AEC-Q100 grade 1.", that is the -`a125` suffix, which does not change that this is sitll `st,m24xxx`. I do not think that even the DTS from kernel with regexes is handling that.`

Yes I think we are all agreeing we don't like the approach in the PR, @benediktibk let's also make sure leave out words and indicators for "automotive" and different standards like @de-nordic is pointing out, just making a generic binding for each vendor family seems reasonable. To summarize I think this PR should just add one binding for the ST M24 eeprom family, and not mention any standards or markets, and add one DT_FOREACH to the existing driver.

**Comment by benediktibk** [comment]
> Yes I think we are all agreeing we don't like the approach in the PR, @benediktibk let's also make sure leave out words and indicators for "automotive" and different standards like @de-nordic is pointing out, just making a generic binding for each vendor family seems reasonable. To summarize I think this PR should just add one binding for the ST M24 eeprom family, and not mention any standards or markets, and add one DT_FOREACH to the existing driver.

With that approach you are still relying on vendors not doing some weird stuff with their part numbers. It might be currently the case that everything in the range of M24xxx works similiarly enough to be handled by one driver, but I wouldn't bet my life on this to be true forever.

Anyway, this is an acceptable approach for me and I've pushed the PR to this state.

**Comment by benediktibk** [comment]
I have a feeling that this is a rather general discussion and topic than an actual PR review. Should we bring this either to the dev-review or the architecture work group?

**Comment by decsny** [comment]
> > Yes I think we are all agreeing we don't like the approach in the PR, @benediktibk let's also make sure leave out words and indicators for "automotive" and different standards like @de-nordic is pointing out, just making a generic binding for each vendor family seems reasonable. To summarize I think this PR should just add one binding for the ST M24 eeprom family, and not mention any standards or markets, and add one DT_FOREACH to the existing driver.
> 
> With that approach you are still relying on vendors not doing some weird stuff with their part numbers. It might be currently the case that everything in the range of M24xxx works similiarly enough to be handled by one driver, but I wouldn't bet my life on this to be true forever.
> 
> Anyway, this is an acceptable approach for me and I've pushed the PR to this state.

Let's just worry about that case if/when we actually see it

**Comment by benediktibk** [comment]
The build error seems to be unrelated to these changes:

     Traceback (most recent call last):
      File "/home/runner/.local/lib/python3.10/site-packages/sphinx/cmd/build.py", line 337, in build_main
        app.build(args.force_all, args.filenames)
      File "/home/runner/.local/lib/python3.10/site-packages/sphinx/application.py", line 351, in build
        self.builder.build_update()
      File "/home/runner/.local/lib/python3.10/site-packages/sphinx/builders/__init__.py", line 293, in build_update
        self.build(to_build,
      File "/home/runner/.local/lib/python3.10/site-packages/sphinx/builders/__init__.py", line 313, in build
        updated_docnames = set(self.read())
      File "/home/runner/.local/lib/python3.10/site-packages/sphinx/builders/__init__.py", line 417, in read
        self._read_parallel(docnames, nproc=self.app.parallel)
      File "/home/runner/.local/lib/python3.10/site-packages/sphinx/builders/__init__.py", line 473, in _read_parallel
        tasks.join()
      File "/home/runner/.local/lib/python3.10/site-packages/sphinx/util/parallel.py", line 107, in join
        if not self._join_one():
      File "/home/runner/.local/lib/python3.10/site-packages/sphinx/util/parallel.py", line 125, in _join_one
        exc, logs, result = pipe.recv()
      File "/usr/lib/python3.10/multiprocessing/connection.py", line 250, in recv
        buf = self._recv_bytes()
      File "/usr/lib/python3.10/multiprocessing/connection.py", line 414, in _recv_bytes
        buf = self._recv(4)
      File "/usr/lib/python3.10/multiprocessing/connection.py", line 383, in _recv
        raise EOFError
    EOFError
    
    Exception occurred:
      File "/usr/lib/python3.10/multiprocessing/connection.py", line 383, in _recv
        raise EOFError

**Comment by henrikbrixandersen** [comment]
> Ok, then, that's basically quite aligned with the content of my review ([here](https://github.com/zephyrproject-rtos/zephyr/pull/73136#discussion_r1609624013) and [here](https://github.com/zephyrproject-rtos/zephyr/pull/73136#discussion_r1609637588)) so I can't agree more.

Please see https://github.com/zephyrproject-rtos/zephyr/pull/73512

File: drivers/eeprom/eeprom_at2x.c
Comment (pdgendt) [inline drivers/eeprom/eeprom_at2x.c:40]: This now also has to check `st_m24xxx` compatibles?

Comment (benediktibk) [reply drivers/eeprom/eeprom_at2x.c:40]: true that

**Review by pdgendt — DISMISSED** [review]
I think this makes sense, but needs maintainer (Brix) approval. Should this still be considered for 3.7?

**Comment by benediktibk** [comment]
> I think this makes sense, but needs maintainer (Brix) approval. Should this still be considered for 3.7?

I don't mind when it is getting merged, I just think it should end up in main at some point. Maybe we can convince @henrikbrixandersen with the next PR which tries to implement an AT24 variant ;-).

**Comment by henrikbrixandersen** [comment]
> > I think this makes sense, but needs maintainer (Brix) approval. Should this still be considered for 3.7?
> 
> I don't mind when it is getting merged, I just think it should end up in main at some point. Maybe we can convince @henrikbrixandersen with the next PR which tries to implement an AT24 variant ;-).

As I have told you already, this PR will not be merged upstream. We already have what you call "official support for the ST M24 I2C EEPROM series". The proper solution is to use the devicetree "compatible" string as it is intended, and simply add a "atmel,at24" (as this is the common programming model for this type of device) at the end of the list..

**Comment by benediktibk** [comment]
Rebased onto main to resolve merge conflicts.

**Pull request closed without merging**
